actors/builtin/miner/miner_actor.go

```diff
@@ -638,37 +635,35 @@ func (a Actor) AwardReward(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
 
 	var st State
 	rt.State().Transaction(&st, func() interface{} {
-		if err := st.addLockedFund(adt.AsStore(rt), rt.CurrEpoch(), pledgeAmount); err != nil {
+		if err := st.AddLockedFunds(adt.AsStore(rt), rt.CurrEpoch(), pledgeAmount, &PledgeVestingSpec); err != nil {
 			rt.Abortf(exitcode.ErrIllegalState, "failed to add pledge: %v", err)
 		}
 		return nil
 	})
 
-	a.updateTotalPledge(rt, &pledgeAmount)
+	a.notifyPledgeChanged(rt, &pledgeAmount)
 	return nil
 }
 
 func (a Actor) slashPledgeCollateral(rt Runtime, amountToSlash abi.TokenAmount) *abi.TokenAmount {
-	if amountToSlash.LessThanEqual(big.Zero()) {
-		rt.Abortf(exitcode.ErrIllegalArgument, "non positive amount to slash: %s", amountToSlash)
-	}
+	AssertMsg(amountToSlash.GreaterThanEqual(big.Zero()), "negative slash amount %s", amountToSlash)
 
 	var st State
-	amountSlashed := rt.State().Transaction(&st, func() interface{} {
-		slashedPledge, err := st.slashLockedFund(adt.AsStore(rt), rt.CurrEpoch(), amountToSlash)
+	slashable := rt.State().Transaction(&st, func() interface{} {
+		// Unlock *unvested* amounts in order to burn them.
+		slashable, err := st.UnlockUnvestedFunds(adt.AsStore(rt), rt.CurrEpoch(), amountToSlash)
 		if err != nil {
```

Comment:
this should never error out?

Reply:
I don't quite understand what you mean, but it's never acceptable to ignore the error. Perhaps especially so if it should "never" happen!